Modules/Core/Common/test/itkImageRegionGTest.cxx

```diff
@@ -32,29 +31,17 @@ TEST(ImageRegion, ZeroSizedRegionIsInside)
     using SizeType = typename RegionType::SizeType;
 
     auto paddedRegion = region;
-    paddedRegion.PadByRadius(1);
+    paddedRegion.PadByRadius(2);
+
+    // Extend the specified region (in 2D, by an extra column).
+    const RegionType extendedRegion(region.GetIndex(), region.GetSize() + SizeType::Filled(1));
```

Comment:
This feels like we have traded one unintended exception for another.

Reply:
Indeed! I only realized that this PR would trade one unintended exception for another from the comment https://github.com/InsightSoftwareConsortium/ITK/pull/3108#issuecomment-1014808455 by @Leengit !

As an alternative, we could reconsider PR #3110 "BUG: region.IsInside(zeroSizedRegion) should always return false" 😸

Reply:
What say we go with PR #3110 instead?

Reply:
We could. Please move the tests over there 😄

Reply:
PR #3110 already has a test, and it's very simple  😄 https://github.com/InsightSoftwareConsortium/ITK/blob/85268b9455bf22973b30fc8755f2a3b56df883be/Modules/Core/Common/test/itkImageRegionGTest.cxx#L24-L25

Reply:
I meant copy the tests which do no involve zero-sized region.

Reply:
@dzenanz Just updated PR #3110 by copying the test for one-sized regions to its GTest. Note that the old-style itkImageRegionTest already had a few tests that involve non-zero-sized regions as well. They still pass successfully (without any adjustment) with either one of the two pull requests, #3108 or #3110.

Reply:
@dzenanz 
> This feels like we have traded one unintended exception for another.

Of course, the unintended exception of this PR could also be fixed by adding an extra check that the index of the inner region (argument) may not be beyond the end of the outer region.  As I mentioned at https://github.com/InsightSoftwareConsortium/ITK/pull/3108#issuecomment-1014906560 ("Another possibility is to add an extra check, to ensure that a zero-sized region is considered outside, when its index is outside"), including a possible implementation. With the extra check, IsInside(zeroSizedRegion) would be true, if and only if IsInside(zeroSizedRegion.GetIndex()) would be true. It would not affect the behavior of non-zero-sized regions.